apollo-router/src/plugins/telemetry/config_new/attributes.rs

```diff
@@ -0,0 +1,702 @@
+    OperationKind {
+        /// The operation kind from the query (query|mutation|subscription).
+        operation_kind: OperationKind,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    QueryVariable {
+        /// The name of a graphql query variable.
+        query_variable: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    ResponseBody {
+        /// Json Path into the response body
+        response_body: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    RequestHeader {
+        /// The name of the request header.
+        request_header: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    ResponseHeader {
+        /// The name of the response header.
+        response_header: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    RequestContext {
+        /// The request context key.
+        request_context: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<AttributeValue>,
+    },
+    ResponseContext {
+        /// The response context key.
+        response_context: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<AttributeValue>,
+    },
+    Baggage {
+        /// The name of the baggage item.
+        baggage: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    Env {
+        /// The name of the environment variable
+        env: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, JsonSchema, Clone, Debug)]
+#[serde(deny_unknown_fields, rename_all = "snake_case", untagged)]
+pub(crate) enum SubgraphCustomAttribute {
+    SubgraphOperationName {
+        /// The operation name from the subgraph query.
+        subgraph_operation_name: OperationName,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    SubgraphOperationKind {
+        /// The kind of the subgraph operation (query|mutation|subscription).
+        subgraph_operation_kind: OperationKind,
+        /// Optional default value.
+        default: Option<String>,
```

Comment:
Do we need a default value for that ? a subgraph operation should always have an operation kind

Reply:
[a7d501c](https://github.com/apollographql/router/pull/4061/commits/a7d501cdb0ffad517b7420e328d01b8dcfd8f0b7)